Some fixes for doc

hkollmann opened the pull request:
- fix links and text in mobile
   - start doc of promise
   - fix server doc

hkollmann committed:
Some fixes:
   - fix links and text in mobile
   - start doc of promise
   - fix server doc

derrell commented on docs/_sidebar.md:
```diff
@@ -157,3 +153,5 @@
     - [7. Translations](/tutorial/twitter/tutorial-part-7.md)
     - [8. Unit Testing](/tutorial/twitter/tutorial-part-8.md)
     - [9. Virtual Lists](/tutorial/twitter/tutorial-part-9.md)
+  - [Qooxdoo Mobile App](/mobile/tutorial.md)
```
All of the items above this new one are numbered. Should not this one be too?

cboulanger replied:
I think it's a different tutorial specifically for mobile apps - the above ones belong to the website version.

cboulanger reviewed (approved):
I did not even know we had setXxxAsync !!

derrell commented on docs/mobile/deployment.md:
```diff
@@ -14,24 +14,24 @@ On the Cordova website you find detailed tutorials which describe the deployment
 ----------------------------------------------------------
 
-After you installed cordova, and followed the introductions of [Cordova Getting Started Tutorial](http://docs.phonegap.com/en/1.6.1/guide_getting-started_index.md.html#Getting%20Started%20Guides), you are able to deploy your %{Mobile} application on your mobile device.
+After you installed cordova, and followed the introductions of [Cordova Getting Started Tutorial](http://docs.phonegap.com/en/1.6.1/guide_getting-started_index.md.html#Getting%20Started%20Guides), you are able to deploy your Mobile application on your mobile device.
```
That documentation is seriously outdated...

derrell replied:
(Also old docs referenced elsewhere herein.)

derrell replied:
"That documentation" meaing the phonegap links

derrell reviewed (changes requested):
I don't know enough about Mobile to comment much on it, but I did have a few comments here...

hkollmann committed:
delete empty pages

cboulanger commented:
I think the heading "Qooxdoo Tools" in the sidebar should really be called "Development" because taht is what this section is really about - how do I develop a qooxdoo application, using CLI, tools, and other specific information. 

To fix to fix https://github.com/qooxdoo/qooxdoo-compiler/issues/530 , we need 
 - to put an entry for ["Customising Application Startup"](http://www.qooxdoo.org/docs/#/compiler/CustomAppStartup)  in this section and
 - to put a minified URL of http://www.qooxdoo.org/docs/#/compiler/CustomAppStartup into https://github.com/qooxdoo/qooxdoo-compiler/blob/master/source/class/qx/tool/compiler/Console.js#L185

cboulanger commented:
@hkollmann Can you intergrate this into your PR?

hkollmann commented:
I am thinking about cleaning up the sidebar. IMHO it's to confusing because of it's length.
Any ideas how to cleanup the widget section?

hkollmann committed:
- fix cordova version
- fix link in sidebar

cboulanger commented:
Create a widget index/overview page containing the stuff that‘s currently in the sidebar....

hkollmann committed:
cleanup widget and layout lists

hkollmann committed:
add all of the compiler docs

hkollmann committed:
revert changes

hkollmann committed:
mark development as to be done

hkollmann committed:
rename to original doc names so that we can better find out which stuff is missing

hkollmann commented:
During working on doc i found some more interesting topics in the old docs which we should rework, e.g. the stuff in documentation\manual\source\pages\data_binding. This is only partly converted to md format. 
I startet this work with renaming the folders to the old name so that it's easier to get an overwiew what's missing.
Help is welcomed!

hkollmann commented:
to be reviewed:
   /pages/core  : missing documents
  /pages/data_binding: missing documents
  /pages/desktop: new Name gui with name changes.
 /pages/widgets: to be integrated
/pages/desktop to be integrated into /gui/README.md + adding missing docs.

+

hkollmann committed:
add missing data binding stuff

hkollmann committed:
clean up structure

hkollmann committed:
add missing core stuff

hkollmann committed:
fix links in core

hkollmann committed:
add widget refs

hkollmann committed:
disable navbar feature

hkollmann committed:
add missing doc to gui

hkollmann committed:
Merge branch 'fix_doc_2' of https://github.com/hkollmann/qooxdoo into fix_doc_2

hkollmann committed:
fix links

cboulanger reviewed (approved):
Thank you very much for this huge effort!!! 🥇

derrell commented on docs/core/oo_feature_summary.md:
```diff
@@ -0,0 +1,387 @@
+    {
+      statics:
+      {
+        driveLetter : "C"
```
Documenting what the defer section will do is very useful for later figuring out where the declarations of variables are located. I would encourage developers to add here:
```
drive : null  // set in defer method
```
 and a members section too:
```
members:
{
  whatsTheDrive : null  // set in defer method
}
```

derrell commented on docs/ebnf_like.md:
```diff
@@ -0,0 +1,14 @@
+| Syntax symbol     | Description                                               |
+| ----------------- | --------------------------------------------------------- |
+| **:=**            | production rule (*"non-terminal can be expanded to ..."*) |
+| **{** ... **}**   | 0..N occurrences (*"\*"*)                                 |
```
All of the EBNF documents above allow for a trailing comma even on the last item. We should at least document, here, that although they are used that way, the expectation is that the final element in a list will not have a trailing comma.

derrell reviewed (changes requested):
I agree with @cboulanger. Herculean effort! Much appreciated.

I noticed a few things that should probably get fixed (or tell me it'll have to do, and I'll accept that)

hkollmann commented:
I just converted the old documentation and added some missing parts. I do not changed the content.
Could we make two PR of this: 
- This one with the old docs.
- A second one with content change?

@derrell : I would be very happy if you could do this.

There some more chapters in the old docs wich are worth to convert. I will do this in a later PR.

derrell commented:
None of my comments are things that must go into this right now, so yes, I'll approve this PR and those changes can go into a future PR.

johnspackman reviewed (approved):
wow, awesome amount of work!!  well done & thank you @hkollmann !!
